tsl/src/continuous_aggs/Readme.md

```diff
@@ -1,144 +1,115 @@
+other buckets, and buckets can be combined to form more granular
+aggregates (e.g., hourly buckets can be combined to form daily
+buckets). Finalization of the partial buckets happens automatically at
+query time. Although such finalization gives slightly higher querying
+times, it is offset by more efficient refreshes that only recompute
+the buckets that have been "invalidated" by changes in the raw data.
+
+A continuous aggregate policy automates the refreshing, allowing the
+aggregate to stay up-to-date without manual intervention. A policy can
+be configured to only refresh the most recent data (e.g., just the
+last hour's worth of data) or ensure that the continuous aggregate is
+always up-to-date with the underlying source data. Policies that focus
+on recent data allow older parts of the continuous aggregate to stay
+the same or be governed by manual refreshes.
+
+## Bookkeeping and Internal State ##
+
+TimescaleDB does bookkeeping for each continuous aggregate to know
+which buckets of the aggregates require refreshing. Whenever a
+modification happens to the source data, an invalidation for the
+modified region is written to an invalidation log. However,
+invalidations are not written after the *invalidation threshold*,
+which tracks the latest bucket materialized thus far. This threshold
+allows write amplification to be kept to a minimum by, e.g, never
+refreshing the most recent time bucket that is still being "filled".
+
+Thus, to store, maintain, and query aggregations, continuous
+aggregates consist of the following objects:
+
+1. A user view, which queries and finalizes the aggregations and is
+   also the object that users interact with.
+2. A partial view, which is used to materialize new data.
+3. A direct view, which holds the original query that users specified.
+4. An internal materialization hypertable, containing the materialized
+   data as partial aggregates for each time bucket.
+5. An invalidation threshold, which tracks how "far" the
+   materialization has reached and below which invalidations must be
+   logged.
+6. A trigger on the source hypertable that invalidates regions of data
+   in every transaction, based on INSERT, UPDATE, and DELETE
+   statements. The trigger writes its invalidations to the hypertable
+   invalidation log.
+7. A hypertable invalidation log that tracks invalidated regions of
+   data for each hypertable. Entries in this log contain time ranges
+   that need to be re-materialized across all the hypertable's
+   continuous aggregates.
+8. A materialization invalidation log. Once a refresh runs on a given
+   continuous aggregate, this log tracks how invalidations from the
+   hypertable invalidation log are processed against the refresh
+   window for the refreshed continuous aggregate. Thus, a single
+   invalidation in the hypertable invalidation log becomes one entry
+   per continuous aggregate in the materialization invalidation log.
```

Comment:
```suggestion
8. A materialization invalidation log for each continuous aggregate. Once a refresh runs on a given
   continuous aggregate, this log tracks how invalidations from the
   hypertable invalidation log are processed against the refresh
   window for the refreshed continuous aggregate. Thus, a single
   invalidation in the hypertable invalidation log becomes one entry
   per continuous aggregate in the materialization invalidation log.
```

Reply:
There is only one log, and your suggestion could be interpreted as if there is one log per cagg.